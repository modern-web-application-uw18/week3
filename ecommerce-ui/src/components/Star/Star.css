.AlignRight {
    text-align: right; 
    width: 100%;
    color: #ffffff;
    opacity: .5;

}

.Star {
    height: 20px;
    width: 20px;
    margin-bottom: 5px;
}

@media (max-width: 668px) {
    .Star {
        height: 12px;
        width: 12px;
        margin-bottom: 2px;
    }

}