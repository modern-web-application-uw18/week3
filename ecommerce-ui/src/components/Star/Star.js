import React from 'react';
import classes from './Star.css';
import MdStar from 'react-icons/lib/md/star';
import MdStarHalf from 'react-icons/lib/md/star-half';
import PropTypes from 'prop-types';




const star = (props) => {

    let starArrMaker = (starCount) => {
        let stars = [];
        if(starCount % 1 !== 0) {stars.push("half")};
        
        while(starCount >= 1) {
            stars.push("whole");
            starCount--;
        }
        return stars.reverse();
    }


    return (
        <div className={classes.AlignRight}>
            {starArrMaker(props.starCount).map((item, idx) => {
                return item === 'whole' ? <MdStar className={classes.Star} key={idx} /> : <MdStarHalf className={classes.Star} key={idx} />
            })}
        </div>

    )
}


star.propTypes = {
    starCount: PropTypes.number.isRequired
}



export default star;