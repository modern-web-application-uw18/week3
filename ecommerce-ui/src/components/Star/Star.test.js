import React from 'react';
import { shallow, mount} from 'enzyme';
import Star from './Star';
import classes from './Star.css';
import MdStar from 'react-icons/lib/md/star';
import MdStarHalf from 'react-icons/lib/md/star-half';

describe('<Star />', () => {
    it('Should have three MdStar elements', () => {
        const wrapper = mount(<Star starCount={3} />);
        expect(wrapper.find(MdStar)).toHaveLength(3);
        expect(wrapper).toMatchSnapshot();

    })
})