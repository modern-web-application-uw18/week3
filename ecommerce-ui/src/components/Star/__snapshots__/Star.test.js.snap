// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Star /> Should have three MdStar elements 1`] = `
<star
  starCount={3}
>
  <div>
    <MdStar
      key="0"
    >
      <IconBase
        viewBox="0 0 40 40"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m20 28.8l-10.3 6.2 2.7-11.7-9-7.9 11.9-1 4.7-11 4.7 11 11.9 1-9 7.9 2.7 11.7z"
            />
          </g>
        </svg>
      </IconBase>
    </MdStar>
    <MdStar
      key="1"
    >
      <IconBase
        viewBox="0 0 40 40"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m20 28.8l-10.3 6.2 2.7-11.7-9-7.9 11.9-1 4.7-11 4.7 11 11.9 1-9 7.9 2.7 11.7z"
            />
          </g>
        </svg>
      </IconBase>
    </MdStar>
    <MdStar
      key="2"
    >
      <IconBase
        viewBox="0 0 40 40"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m20 28.8l-10.3 6.2 2.7-11.7-9-7.9 11.9-1 4.7-11 4.7 11 11.9 1-9 7.9 2.7 11.7z"
            />
          </g>
        </svg>
      </IconBase>
    </MdStar>
  </div>
</star>
`;
